kubernetes-client/src/com/goyeau/kubernetes/client/api/PodsApi.scala

```diff
@@ -71,50 +96,190 @@ private[client] class NamespacedPodsApi[F[_]](
+      stderr: Boolean = true,
+      tty: Boolean = false
+  ): Resource[F, F[Stream[F, Either[ExecStream, ErrorOrStatus]]]] = {
+    val request = execRequest(podName, command, container, stdin, stdout, stderr, tty)
+
+    wsClient.connectHighLevel(request).map { connection =>
+      F.delay(connection.receiveStream.through(processWebSocketData))
     }
   }
 
-  // first byte defines which stream incoming message belongs to
-  private def processWebSocketData: Pipe[F, WSDataFrame, (List[ExecStream], Either[String, Status])] =
+  def exec(
+      podName: String,
+      container: Option[String] = None,
+      command: Seq[String] = Seq.empty,
+      stdin: Boolean = false,
+      stdout: Boolean = true,
+      stderr: Boolean = true,
+      tty: Boolean = false
+  ): F[(List[ExecStream], Option[ErrorOrStatus])] =
+    execStream(podName, container, command, stdin, stdout, stderr, tty).use(_.flatMap(foldStream))
+
+  private def foldStream(
+      stdoutStream: Stream[F, Either[ExecStream, ErrorOrStatus]]
+  ) =
+    stdoutStream.compile.fold((List.empty[ExecStream], none[ErrorOrStatus])) { case ((accEvents, accStatus), data) =>
+      data match {
+        case Left(event) =>
+          (accEvents :+ event) -> accStatus
+        case Right(errorOrStatus) =>
+          accEvents -> accStatus.orElse(errorOrStatus.some)
+      }
+    }
+
+  private def foldErrorStream(
+      stdoutStream: Stream[F, Either[ExecStream, ErrorOrStatus]]
+  ) =
+    stdoutStream.compile.fold((List.empty[StdErr], none[ErrorOrStatus])) { case ((accEvents, accStatus), data) =>
+      data match {
+        case Left(event) =>
+          accEvents ++ (event match {
+            case e: StdErr => e.some
+            case _         => none[StdErr]
+          }).toList -> accStatus
+        case Right(errorOrStatus) =>
+          accEvents -> accStatus.orElse(errorOrStatus.some)
+      }
+    }
+
+  private def processWebSocketData: Pipe[F, WSDataFrame, Either[ExecStream, ErrorOrStatus]] =
     _.collect {
-      case WSFrame.Binary(data, _) if data.headOption.contains(1.toByte) =>
-        List(StdOut(convertToString(dropRoutingByte(data)))) -> "".asLeft
-      case WSFrame.Binary(data, _) if data.headOption.contains(2.toByte) =>
-        List(StdErr(convertToString(dropRoutingByte(data)))) -> "".asLeft
-      case WSFrame.Binary(data, _) if data.headOption.contains(3.toByte) =>
+      case WSFrame.Binary(data, _) if data.headOption.contains(StdOutId) =>
+        StdOut(dropRoutingByte(data).toArray).asLeft
+
+      case WSFrame.Binary(data, _) if data.headOption.contains(StdErrId) =>
+        StdErr(dropRoutingByte(data).toArray).asLeft
+
+      case WSFrame.Binary(data, _) if data.headOption.contains(StatusId) =>
         val json = convertToString(dropRoutingByte(data))
-        Nil -> decode[Status](json).left.flatMap(_ => json.asLeft[Status])
+        decode[Status](json).leftFlatMap(_ => ParseFailure(json).asLeft[Status]).asRight[ExecStream]
```

Comment:
There are 3 cases for Status:
1. It can be missing for certain operation
2. It can be parsed with issue, if client has old/incompatible case class models versus current k8s api version
3. It can be parsed successfully